docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
+![Stock.jpeg](UML%20diagram%2FStock.jpeg)
 
+The mechanism:
+1. `stock` command invoke the `StockCommand` class.
+2. `StockCommand` invokes the `start` method for `StockActivate` class.
+3. `StockActivate` class calls a new `StockStorage` class.
+4. `StockStorage` class will new execute its play function and enable user to buy stocks.
+
+Here is the sequence diagram of `sellstock` command:
+
+![SellStock.jpeg](UML%20diagram%2FSellStock.jpeg)
+
+The mechanism:
+1. `sellstock` command invoke the `SellStockCommand` class.
+2. `SellStockCommand` invokes the `getAsset` method for `PlayerProfile` and fetch the `Asset` class.
+3. `Asset` class executes the `sellStock()` function and sell all of the stocks the player currently possess.
+4. Information related to stocks will be updated inside `Asset` and `PlayerProfile` class.
+
+![Save.png](UML%20diagram%2FSave.png)
+
+The mechanism:
+1. `Saver` will be called automatically after one command is executed.
+2. `Saver` creates a new `file` class and invokes the `constructJson` method for `PlayerProfile` class.
+3. `Saver` class calls `write` and `flush` method from the `file` class to write to the json file.
+
+![Load.png](UML%20diagram%2FLoad.png)
+
+The mechanism:
+1. `Loader` will be called automatically once the program is initialized.
+2. `Loader` invokes the `readJsonFromFile` method to read data from the json file and creates a new `file` class.
+3. `Loader` invokes the `parseJson` method to retrieve relevant attributes and creates a new `PlayerProfile` class.
+4. `Loader` returns the `PlayerProfile` class to load the previous information.
+
+
+## MiniGame components
+
+Here is the partial class diagram of the `MiniGame` component:
+
+![MiniGame.png](UML%20diagram%2FMiniGame.png)
+
+The `MiniGame` mechanism:
+1. For command `work`, `rest`, and `exercise`, they would have their respective mini-games.
+2. When these commands are generated and executed in `EconoCraftLogic`, the respective mini-game would be played.
+3. The command would then update the player profile according to the mini-game result.
+
+# Implementation
+
+## MiniGame - Typing Game
+
+The implementation of the Typing Game is as follows:
+
+1. The game can be invoked by the `WorkCommand` class when the user inputs the `work` command.
+2. It makes use of the `ResponseManager` to generate the instructions of the game to the user. 
+3. The user would be prompted to type the given text as fast as they can.
+4. This game made use of the `CompletableFuture` class to create separate thread which handles the countdown timer and user input at the same time.
+5. When the user finishes typing or the time limit is reached, the game would calculate the user's typing speed and accuracy and reward the user accordingly. 
+6. Finally, the `WorkCommand` would update the player profile with the reward earned according to the user's performance in the game.
+
+## MiniGame - Tic Tac Toe
+
```

Comment:
Perhaps some diagrams to represent the architectures of the mini games would be nice.